service/state/encoding.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package state
+
+import "encoding/binary"
+
+func Encode(pins ...interface{}) []byte {
```

Comment:
I find `pins` confusing, because the first thing imo people think about is the PIN for the phone or the credit card, not pins of a lock. You could simply use `parts` here. Or `segments`?

Reply:
I would prefer if we actually use `Encode(prefix uint8, segments ...interface{})` here. A key should *always* start with a prefix, so that would make it statically checked. We could then also remove the `uint8` case from the type switch.

I would also argue that we should add `string` support here explicitly. As we allow `interface{}`, it will be less annoying to use the function that way.

Reply:
Excellent suggestions. How would you deal with strings though? We transform it into a hex-encoded byte slice, right? Or do we consider that it's already hex-encoded?

Reply:
That's a good point, let's just ignore strings for now, I think we need to discuss it.